scripts/check_types.py

```diff
@@ -0,0 +1,162 @@
+        error_path = Path(error["file"])
+        try:
+            rel_path = error_path.relative_to(base_dir)
+            error_files.add(str(rel_path))
+        except ValueError:
+            # If the file is not relative to base_dir, use the full path
+            error_files.add(str(error_path))
+    return error_files
+
+
+def build_file_tree(files, base_dir, error_files):
+    """Build a tree structure of files for display."""
+    tree = defaultdict(lambda: defaultdict(list))
+    # Get files with no functions
+    empty_files = []
+    for file in files:
+        if not check_file_for_functions(file):
+            empty_files.append(str(file.relative_to(base_dir)))
+    for path in sorted(files):
+        rel_path = path.relative_to(base_dir)
+        parts = rel_path.parts
+        subdir = str(Path(*parts[:-1])) if len(parts) > 1 else ""
+        rel_path_str = str(rel_path)
+        # Determine status - empty, typed, or untyped
+        if rel_path_str in empty_files:
+            status = "empty"
+        else:
+            status = "typed" if rel_path_str not in error_files else "untyped"
+        tree[base_dir][subdir].append((parts[-1], status))
+    return tree, len(empty_files)
+
+
+def print_tree(tree):
+    """Print the tree of files with type hint status."""
+    # here intentionally using print here instead of logger
+    # to output the tree to stdout as requested
+    print("\nType Hints Status:")  # noqa: T201
+    print(".")  # noqa: T201
+    for base_dir in sorted(tree):
+        print(f"├── {Path(base_dir).name}")  # noqa: T201
+        for _, (subdir, files_in_dir) in enumerate(sorted(tree[base_dir].items())):
+            if subdir:
+                print(f"│   ├── {subdir}")  # noqa: T201
+                prefix = "│   │   "
+            else:
+                prefix = "│   "
+            for j, (filename, status) in enumerate(sorted(files_in_dir)):
+                is_last = j == len(files_in_dir) - 1
+                if status == "empty":
+                    checkbox = "[e]"
+                elif status == "typed":
+                    checkbox = "[x]"
+                else:
+                    checkbox = "[ ]"
+                print(  # noqa: T201
+                    f"{prefix}{'└── ' if is_last else '├── '}{checkbox} {filename}"
+                )
+
+
+def print_summary(typed, total, empty):
+    """Print summary statistics."""
+    files_with_functions = total - empty
+    percentage = (typed / files_with_functions * 100) if files_with_functions > 0 else 0
+    print(  # noqa: T201
+        f"\nSummary: {typed}/{files_with_functions} files with functions have complete type hints ({percentage:.1f}%)"
+    )
+    if empty > 0:
+        print(f"         {empty} files have no function definitions")  # noqa: T201
+
+
+def main():
+    # Setup
+    script_path = Path(__file__).resolve().parent
+    paths = sys.argv[1:] or [
+        script_path.parent / "prolif",
+        script_path.parent / "tests",
+    ]
```

Comment:
I had an issue running this script when specifying one specific file instead of using the defaults:
```
ValueError: 'tests/test_residues.py' is not in the subpath of '/Users/cbouy/dev/ProLIF' OR one path is relative and the other is absolute.
```
If you get rid of the `resolve()` part and make the script path relative instead it works:
```suggestion
    base_dir = Path()
    paths = sys.argv[1:] or [
        base_dir / "prolif",
        base_dir / "tests",
    ]
```
and you can remove the `base_dir = script_path.parent.resolve()` after the mypy command.
This script is supposed to be ran from the root directory anyway.